tmui/src/app/src/core/tool/LabelLayer.ts

```diff
@@ -77,10 +77,13 @@ abstract class LabelLayer extends VectorTileLayer {
             var geomType = feature.getGeometry().getType();
             var label = feature.get('label');
             var fillColor: ol.Color;
-            if (label !== undefined) {
-                fillColor = this.colorMapper(label).toOlColor();
+            if (label == "None") {
+                // If no label is provided, color the cell in black
+                // This string assignment results in an error when using 
+                // gulp --prod, but it still works fine in the interface. 
+                fillColor = '#000000';
```

Comment:
Probably here you can use the static value defined here:
https://github.com/pelkmanslab/TissueMAPS/blob/071f3cfd0ff9f52af7d18645ba59bb1b00fd6dcb/tmui/src/app/src/core/util/Color.ts#L161-L163
and set it as "default": this probably could remove the gulp error at compile time.
If then the value is undefined, use the label to extract the color.

```suggestion
             var fillColor: ol.Color.BLACK;
             if (label === undefined) {
                fillColor = this.colorMapper(label).toOlColor();
             } 
```
(I wasn't able to create a multiline patch so you may need to edit it manually)

Reply:
I tried this, but when I use `var fillColor: ol.Color.BLACK;,` I get `app/src/core/tool/LabelLayer.ts(79,31): error TS2305: Module 'ol' has no exported member 'Color'.`
I don't know what this 'ol' is to be honest or where it is defined...
I also tried directly using `fillColor = new Color(0, 0, 0);`, but then I get `app/src/core/tool/LabelLayer.ts(82,17): error TS2322: Type 'Color' is not assignable to type 'ol.Color'.` So we seem to have different kind of color classes here...

Also, unfortunately, I can't check for undefined. Took me half an hour to figure this out, but it's actually the exact string "None", not a null or other undefined value that is given for undefined values here.

Reply:
For reference, with my original code here, I get `app/src/core/tool/LabelLayer.ts(84,17): error TS2322: Type 'string' is not assignable to type 'Color'.`

Reply:
@jluethi and what if you try `Color.BLACK.toOlColor()` instead of `ol.Color.BLACK` as I initially suggested?

Reply:
That worked!! :) Thanks @bebosudo !